backend/Application/Aggregates/Contract/Jobs/_06_AddTokenAddress.cs

```diff
@@ -50,18 +51,20 @@ await Policies.GetTransientPolicy(GetUniqueIdentifier(), _logger, _options.Retry
             .ExecuteAsync(async () =>
             {
                 await using var context = await _contextFactory.CreateDbContextAsync(token);
+                var connection = context.Database.GetDbConnection();
                 var tokens = await context.Tokens
-                    .Where(t => (int)t.ContractIndex == identifier)
+                    .AsNoTracking()
+                    .Where(t => (int)t.ContractIndex == identifier && t.TokenAddress == null)
                     .ToListAsync(token);
+                
                 foreach (var cisToken in tokens.Where(cisToken => cisToken.TokenAddress == null))
```

Comment:
Is the Where statement needed here? 
From my understanding, it is already checked when fetching above this line